src/routes.rs

```diff
@@ -121,6 +128,22 @@ impl IntoResponse for DispenseInfoResponse {
     }
 }
 
+async fn has_reached_daily_limit(address: Address) -> bool {
+    LAST_DISPENSED.with_borrow_mut(|last_dispensed| {
+        let current_time = tokio::time::Instant::now();
+
+        // evict entries older than a day
+        last_dispensed.retain(|_, timestamp| current_time.duration_since(*timestamp) < DAY);
```

Comment:
Maybe it is better to use a combination of hash map and queue to avoid iteration over all addresses. In the case of 1M addresses, each new one will require 1M iterations.

But it is optional for you, because this situation is unlikely =)